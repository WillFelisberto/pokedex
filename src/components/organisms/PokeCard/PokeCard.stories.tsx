import type { Meta, StoryObj } from '@storybook/react';

import { PokeCard } from './PokeCard';

const meta: Meta<typeof PokeCard> = {
  title: 'Organisms/PokeCard', // Categoria e nome do componente
  component: PokeCard, // Componente sendo documentado
  tags: ['autodocs'], // Adiciona documentação automática
  args: {
    // Valores padrão para as props
    pokemon: {
      baseStats: {
        hp: 45,
        attack: 49,
        defense: 49,
        specialAttack: 65,
        specialDefense: 65,
        speed: 45
      },
      japaneseName: 'イビス',
      region: 'Kanto',
      id: 2,
      name: 'ivysaur',
      sprites: {
        front_default:
          'https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/other/official-artwork/2.png'
      },
      types: ['grass', 'poison'],
      category: 'ivysaur',
      weight: 130,
      height: 10,
      weaknesses: ['flying', 'poison', 'bug', 'fire', 'ice'],
      evolutions: [
        {
          id: 1,
          name: 'bulbasaur',
          url: 'https://pokeapi.co/api/v2/pokemon/1/',
          sprite: 'https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/1.png',
          types: ['grass', 'poison'],
          level: null
        },
        {
          id: 2,
          name: 'ivysaur',
          url: 'https://pokeapi.co/api/v2/pokemon/2/',
          sprite: 'https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/2.png',
          types: ['grass', 'poison'],
          level: 16
        },
        {
          id: 3,
          name: 'venusaur',
          url: 'https://pokeapi.co/api/v2/pokemon/3/',
          sprite: 'https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/3.png',
          types: ['grass', 'poison'],
          level: 32
        }
      ],
      abilities: [
        { name: 'overgrow', is_hidden: false },
        { name: 'chlorophyll', is_hidden: true }
      ],
      genderRatio: {
        male: 87.5,
        female: 12.5
      }
    }
  }
};

export default meta;

// Tipo para as histórias
type Story = StoryObj<typeof PokeCard>;

// História padrão
export const Default: Story = {};

// História sem evoluções
export const SemEvolucoes: Story = {
  args: {
    pokemon: {
      baseStats: {
        hp: 45,
        attack: 49,
        defense: 49,
        specialAttack: 65,
        specialDefense: 65,
        speed: 45
      },
      japaneseName: 'イビス',
      region: 'Kanto',
      ...(meta.args?.pokemon ?? {}),
      evolutions: [], // Remove as evoluções
      id: 2, // Ensure id is defined
      name: 'ivysaur', // Ensure name is defined
      sprites: {
        front_default:
          'https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/other/official-artwork/2.png'
      },
      types: ['grass', 'poison'],
      category: 'ivysaur',
      weight: 130,
      height: 10,
      weaknesses: ['flying', 'poison', 'bug', 'fire', 'ice'],
      abilities: [
        { name: 'overgrow', is_hidden: false },
        { name: 'chlorophyll', is_hidden: true }
      ],
      genderRatio: {
        male: 87.5,
        female: 12.5
      }
    }
  }
};

// História com um Pokémon diferente
export const Charmander: Story = {
  args: {
    pokemon: {
      baseStats: {
        hp: 45,
        attack: 49,
        defense: 49,
        specialAttack: 65,
        specialDefense: 65,
        speed: 45
      },
      japaneseName: 'イビス',
      region: 'Kanto',
      id: 4,
      name: 'charmander',
      sprites: {
        front_default:
          'https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/other/official-artwork/4.png'
      },
      types: ['fire'],
      category: 'charmander',
      weight: 85,
      height: 6,
      weaknesses: ['water', 'ground', 'rock'],
      evolutions: [
        {
          id: 4,
          name: 'charmander',
          url: 'https://pokeapi.co/api/v2/pokemon/4/',
          sprite: 'https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/4.png',
          types: ['fire'],
          level: null
        },
        {
          id: 5,
          name: 'charmeleon',
          url: 'https://pokeapi.co/api/v2/pokemon/5/',
          sprite: 'https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/5.png',
          types: ['fire'],
          level: 16
        },
        {
          id: 6,
          name: 'charizard',
          url: 'https://pokeapi.co/api/v2/pokemon/6/',
          sprite: 'https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/6.png',
          types: ['fire', 'flying'],
          level: 36
        }
      ],
      abilities: [
        { name: 'blaze', is_hidden: false },
        { name: 'solar-power', is_hidden: true }
      ],
      genderRatio: {
        male: 87.5,
        female: 12.5
      }
    }
  }
};
