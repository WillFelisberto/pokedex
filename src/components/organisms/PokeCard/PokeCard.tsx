import Link from 'next/link';
import React from 'react';

import { PokeBadge } from '@/components/atoms/PokeBadge';
import { PokemonHeader } from '@/components/atoms/PokemonHeader';
import { PokemonStatus } from '@/components/molecules/PokemonStatus';

import { PokemonType } from '@/lib/utils/poketypes';
import { formatPokemonId } from '@/lib/utils/StatusConvert';

export interface PokemonProps {
  id: number;
  name: string;
  japaneseName?: string;
  region: string;
  sprites: {
    front_default: string;
  };
  types: PokemonType[];
  category: string;
  weight: number;
  height: number;
  weaknesses: PokemonType[];
  evolutions: {
    id: number;
    name: string;
    url: string;
    sprite: string;
    types: PokemonType[];
    level: number | null;
  }[];
  abilities: { name: string; is_hidden: boolean }[];
  genderRatio: {
    male: number;
    female: number;
  };
  baseStats: {
    hp: number;
    attack: number;
    defense: number;
    specialAttack: number;
    specialDefense: number;
    speed: number;
  };
}

export const PokeCard: React.FC<{ pokemon: PokemonProps }> = ({ pokemon }) => {
  return (
    <div
      data-testid="poke-card"
      key={pokemon.id}
      className="bg-white min-w-[360px] max-w-[360px] p-2 rounded-lg text-center flex flex-col gap-4"
    >
      <Link href={`/pokemon/${pokemon.id}`}>
        <PokemonHeader
          name={pokemon.name}
          spriteUrl={pokemon.sprites.front_default}
          types={pokemon.types}
        />
        <div className="p-2 text-left">
          <h2 className="text-black font-bold capitalize text-3xl" data-testid="pokemon-name">
            {pokemon.name}
          </h2>
          <p
            className="text-base text-gray-700"
            data-testid="pokemon-id"
          >{`NÂº${formatPokemonId(pokemon.id)}`}</p>
        </div>
        <div className="p-2 flex gap-2" data-testid="pokemon-types">
          {pokemon.types.map((type) => (
            <PokeBadge key={`${type}-${pokemon.id}`} type={type} />
          ))}
        </div>
        <div className="p-2 w-full flex gap-[20px]">
          <PokemonStatus height={pokemon.height} weight={pokemon.weight} />
        </div>
      </Link>
    </div>
  );
};
