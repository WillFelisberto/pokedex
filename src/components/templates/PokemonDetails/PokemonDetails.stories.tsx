import type { Meta, StoryObj } from '@storybook/react';
import { QueryClient, QueryClientProvider } from '@tanstack/react-query';

import { PokemonDetails } from './PokemonDetails';
const queryClient = new QueryClient();

const meta: Meta<typeof PokemonDetails> = {
  title: 'Templates/PokemonDetails',
  component: PokemonDetails,
  decorators: [
    (Story) => (
      <QueryClientProvider client={queryClient}>
        <Story />
      </QueryClientProvider>
    )
  ],
  tags: ['autodocs'],
  args: {
    id: 1,
    name: 'bulbasaur',
    japaneseName: 'フシギダネ',
    region: 'Kanto',
    height: 70,
    weight: 69,
    totalPages: 5,
    sprites: {
      front_default:
        'https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/other/official-artwork/1.png'
    },
    types: ['grass', 'poison'],
    baseStats: {
      hp: 45,
      attack: 49,
      defense: 49,
      specialAttack: 65,
      specialDefense: 65,
      speed: 45
    }
  }
};

export default meta;

type Story = StoryObj<typeof PokemonDetails>;

export const Default: Story = {};
