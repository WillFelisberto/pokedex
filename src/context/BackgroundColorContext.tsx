'use client';

import { createContext, useContext, useState } from 'react';

const BackgroundColorContext = createContext({
  backgroundColor: '#ffffff',
  setBackgroundColor: (_color: string) => {}
});

export function BackgroundColorProvider({ children }: { children: React.ReactNode }) {
  const [backgroundColor, setBackgroundColor] = useState('#ffffff');

  return (
    <BackgroundColorContext.Provider value={{ backgroundColor, setBackgroundColor }}>
      <div style={{ backgroundColor, minHeight: '100vh', alignContent: 'center' }}>{children}</div>
    </BackgroundColorContext.Provider>
  );
}

export function useBackgroundColor() {
  return useContext(BackgroundColorContext);
}
