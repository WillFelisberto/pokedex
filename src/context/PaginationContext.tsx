'use client';
import { createContext, useContext, useState, ReactNode } from 'react';

interface PaginationContextType {
  currentPage: number;
  setCurrentPage: (page: number) => void;
}

const PaginationContext = createContext<PaginationContextType | null>(null);

interface PaginationProviderProps {
  children: ReactNode;
}

export const PaginationProvider: React.FC<PaginationProviderProps> = ({ children }) => {
  const [currentPage, setCurrentPage] = useState<number>(1);

  return (
    <PaginationContext.Provider value={{ currentPage, setCurrentPage }}>
      {children}
    </PaginationContext.Provider>
  );
};

export const usePagination = (): PaginationContextType => {
  const context = useContext(PaginationContext);
  if (!context) {
    throw new Error('usePagination must be used within a PaginationProvider');
  }

  return context;
};
