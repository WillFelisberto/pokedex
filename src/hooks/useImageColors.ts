import { useBackgroundColor } from '@/context/BackgroundColorContext';
import { Vibrant } from 'node-vibrant/browser';
import { useState, useEffect } from 'react';

export const useImageColors = (imageUrl: string) => {
  const { setBackgroundColor } = useBackgroundColor();
  const [colors, setColors] = useState<{
    vibrant?: string;
    muted?: string;
    darkVibrant?: string;
    darkMuted?: string;
    lightVibrant?: string;
    lightMuted?: string;
  }>({});

  const [isLoading, setIsLoading] = useState(true); // Estado de carregamento

  useEffect(() => {
    if (!imageUrl) return;

    const fetchColors = async () => {
      setIsLoading(true); // Inicia o loading antes de buscar as cores
      try {
        const palette = await Vibrant.from(imageUrl).getPalette();
        const extractedColor = palette.DarkMuted?.hex || palette.Muted?.hex || '#ffffff';

        setColors({
          vibrant: palette.Vibrant?.hex,
          muted: palette.Muted?.hex,
          darkVibrant: palette.DarkVibrant?.hex,
          darkMuted: palette.DarkMuted?.hex,
          lightVibrant: palette.LightVibrant?.hex,
          lightMuted: palette.LightMuted?.hex
        });

        setBackgroundColor(extractedColor); // Atualiza o fundo dinamicamente
      } catch (error) {
        console.error('Erro ao extrair cores:', error);
      } finally {
        setIsLoading(false); // Finaliza o loading após a execução
      }
    };

    fetchColors();
  }, [imageUrl, setBackgroundColor]);

  return { colors, isLoading };
};
