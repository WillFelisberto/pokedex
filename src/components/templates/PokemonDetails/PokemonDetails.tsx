'use client';

import { useImageColors } from '@/hooks/useImageColors';
import { usePokemonDetails } from '@/hooks/usePokemonDetails';
import clsx from 'clsx';
import Image from 'next/image';

import { Loader } from '@/components/atoms/Loader';
import { Pagination } from '@/components/atoms/Pagination';
import { PokeBadge } from '@/components/atoms/PokeBadge';
import { StatsPill } from '@/components/atoms/StatsPill';
import { PokemonProps } from '@/components/organisms/PokeCard';

export type PokemonDetailsProps = {
  totalPages: number;
} & Pick<
  PokemonProps,
  | 'sprites'
  | 'types'
  | 'baseStats'
  | 'height'
  | 'weight'
  | 'id'
  | 'name'
  | 'japaneseName'
  | 'region'
>;

export const PokemonDetails = ({
  baseStats,
  height,
  id,
  name,
  region,
  sprites,
  types,
  weight,
  japaneseName,
  totalPages
}: PokemonDetailsProps) => {
  const { isLoading } = useImageColors(sprites.front_default);
  const { pokemon, isFetching, goToPage, currentPage } = usePokemonDetails();

  if (isFetching || !pokemon) return <Loader />;

  return (
    <>
      {isLoading && <Loader />}
      <section className="max-w-7xl mx-auto px-4 flex flex-col gap-4 justify-center min-h-screen flex-grow  ">
        <section className="flex flex-col gap-1">
          <p className="text-lg font-bold">{`#${id}`}</p>
          <h1 className="text-4xl font-bold capitalize">{name}</h1>
        </section>
        <div className="flex flex-col lg:flex-row md:flex-col sm:flex-col">
          <section className="flex w-[90%] flex-col lg:flex-row justify-between w-full gap-4">
            {/* Region */}
            <div className="w-full lg:w-[10%] flex-shrink-0 flex self-center">
              <p className="lg:-rotate-90 text-nowrap">Region: {region}</p>
            </div>

            {/* Image and Infos */}
            <div className="w-full lg:w-[60%] flex-shrink-0 relative flex flex-col lg:flex-row items-center lg:items-start">
              <span className="font-bold opacity-20 text-9xl absolute lg:top-0 hidden lg:block">
                {japaneseName}
              </span>

              <div className="flex flex-col mt-4 self-start lg:self-center lg:mt-0 lg:ml-8">
                <span className="flex gap-2">
                  <strong>Height:</strong>
                  <p>{`${height / 10}m`}</p>
                </span>
                <span className="flex gap-2">
                  <strong>Weight:</strong>
                  <p>{`${weight / 10}kg`}</p>
                </span>
              </div>

              <Image
                className="place-self-center z-10 relative"
                src={sprites.front_default}
                alt={name}
                width={400}
                height={400}
              />

              <span className="top-1/2 left-1/2 transform -translate-x-1/2 -translate-y-1/2 font-bold text-center opacity-20 text-9xl absolute  block lg:hidden">
                {japaneseName}
              </span>
            </div>

            {/* Types and Base Stats */}
            <div className="w-full lg:w-[25%] flex flex-col flex-shrink-0 gap-4">
              <div className="flex gap-2 flex-wrap">
                {types.map((type) => (
                  <PokeBadge
                    key={type}
                    className="p-4 rounded-full transition hover:scale-110"
                    badgeType="list"
                    type={type}
                  />
                ))}
              </div>

              <div className="flex flex-col gap-4">
                <h2 className="text-3xl font-bold drop-shadow-2xl">Base Stats:</h2>
                <div className="flex flex-wrap gap-2 border-l-4 border-white/40 px-4">
                  <StatsPill title={`HP: ${baseStats.hp}`} />
                  <StatsPill title={`Attack: ${baseStats.attack}`} />
                  <StatsPill title={`Defense: ${baseStats.defense}`} />
                  <StatsPill title={`SP. Attack: ${baseStats.specialAttack}`} />
                  <StatsPill title={`SP. Defense: ${baseStats.specialDefense}`} />
                  <StatsPill title={`Speed: ${baseStats.speed}`} />
                </div>
              </div>
            </div>
          </section>
          <div
            className={clsx(
              'w-fit flex justify-center self-center',
              'md:flex-row lg:flex-col lg:absolute lg:top-0 lg:right-0'
            )}
          >
            <Pagination
              currentPage={currentPage}
              totalPages={totalPages}
              goToPage={goToPage}
              isFetching={false}
            />
          </div>
        </div>
      </section>
    </>
  );
};
