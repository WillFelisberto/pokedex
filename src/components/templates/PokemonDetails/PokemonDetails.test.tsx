import { PaginationProvider } from '@/context/PaginationContext';
import { useImageColors } from '@/hooks/useImageColors';
import { render, screen } from '@/tests/test-utils';

import { PokemonProps } from '@/components/organisms/PokeCard';

import { PokemonDetails } from './PokemonDetails';

jest.mock('@/hooks/useImageColors', () => ({
  useImageColors: jest.fn()
}));

const mockPokemon: Pick<
  PokemonProps,
  | 'sprites'
  | 'types'
  | 'baseStats'
  | 'height'
  | 'weight'
  | 'id'
  | 'name'
  | 'japaneseName'
  | 'region'
> = {
  id: 25,
  name: 'pikachu',
  japaneseName: 'ピカチュウ',
  region: 'Kanto',
  sprites: { front_default: '/pikachu.png' },
  types: ['electric'],
  height: 40,
  weight: 60,
  baseStats: {
    hp: 35,
    attack: 55,
    defense: 40,
    specialAttack: 50,
    specialDefense: 50,
    speed: 90
  }
};

const renderWithProvider = (ui: React.ReactNode) => {
  return render(<PaginationProvider>{ui}</PaginationProvider>);
};

describe('<PokemonDetails />', () => {
  it('should render the Pokémon details correctly', () => {
    (useImageColors as jest.Mock).mockReturnValue({ isLoading: false });

    renderWithProvider(<PokemonDetails {...mockPokemon} />);

    expect(screen.getByText(`#${mockPokemon.id}`)).toBeInTheDocument();
    expect(screen.getByText(mockPokemon.name)).toBeInTheDocument();
    if (mockPokemon.japaneseName) {
      expect(screen.getByText(mockPokemon.japaneseName)).toBeInTheDocument();
    }

    const image = screen.getByAltText(mockPokemon.name);
    expect(image).toBeInTheDocument();
    expect(image).toHaveAttribute('src', mockPokemon.sprites.front_default);

    mockPokemon.types.forEach((type) => {
      expect(screen.getByAltText(type)).toBeInTheDocument();
    });

    expect(screen.getByText(`HP: ${mockPokemon.baseStats.hp}`)).toBeInTheDocument();
    expect(screen.getByText(`Attack: ${mockPokemon.baseStats.attack}`)).toBeInTheDocument();
    expect(screen.getByText(`Defense: ${mockPokemon.baseStats.defense}`)).toBeInTheDocument();
    expect(
      screen.getByText(`SP. Attack: ${mockPokemon.baseStats.specialAttack}`)
    ).toBeInTheDocument();
    expect(
      screen.getByText(`SP. Defense: ${mockPokemon.baseStats.specialDefense}`)
    ).toBeInTheDocument();
    expect(screen.getByText(`Speed: ${mockPokemon.baseStats.speed}`)).toBeInTheDocument();
  });

  it('should show the loader when isLoading is true', () => {
    (useImageColors as jest.Mock).mockReturnValue({ isLoading: true });

    renderWithProvider(<PokemonDetails {...mockPokemon} />);

    expect(screen.getByAltText('Loading...')).toBeInTheDocument();
  });

  it('should not show the loader when isLoading is false', () => {
    (useImageColors as jest.Mock).mockReturnValue({ isLoading: false });

    renderWithProvider(<PokemonDetails {...mockPokemon} />);

    expect(screen.queryByAltText('Loading...')).not.toBeInTheDocument();
  });
});
