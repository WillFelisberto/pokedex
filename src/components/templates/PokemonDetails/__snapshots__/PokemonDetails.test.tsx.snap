// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PokemonDetails /> should render the Pokémon details correctly 1`] = `
<div>
  <section
    class="max-w-7xl mx-auto px-4 flex flex-col gap-4 justify-center min-h-screen flex-grow  "
  >
    <section
      class="flex flex-col gap-1"
    >
      <p
        class="text-lg font-bold"
      >
        #25
      </p>
      <h1
        class="text-4xl font-bold capitalize"
      >
        pikachu
      </h1>
    </section>
    <div
      class="flex flex-col lg:flex-row md:flex-col sm:flex-col"
    >
      <section
        class="flex w-[90%] flex-col lg:flex-row justify-between w-full gap-4"
      >
        <div
          class="w-full lg:w-[10%] flex-shrink-0 flex self-center"
        >
          <p
            class="lg:-rotate-90 text-nowrap"
          >
            Region: 
            Kanto
          </p>
        </div>
        <div
          class="w-full lg:w-[60%] flex-shrink-0 relative flex flex-col lg:flex-row items-center lg:items-start"
        >
          <span
            class="font-bold opacity-20 text-9xl absolute lg:top-0 hidden lg:block"
          >
            ピカチュウ
          </span>
          <div
            class="flex flex-col mt-4 self-start lg:self-center lg:mt-0 lg:ml-8"
          >
            <span
              class="flex gap-2"
            >
              <strong>
                Height:
              </strong>
              <p>
                4m
              </p>
            </span>
            <span
              class="flex gap-2"
            >
              <strong>
                Weight:
              </strong>
              <p>
                6kg
              </p>
            </span>
          </div>
          <img
            alt="pikachu"
            class="place-self-center z-10 relative"
            height="400"
            src="https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/other/official-artwork/25.png"
            width="400"
          />
          <span
            class="top-1/2 left-1/2 transform -translate-x-1/2 -translate-y-1/2 font-bold text-center opacity-20 text-9xl absolute  block lg:hidden"
          >
            ピカチュウ
          </span>
        </div>
        <div
          class="w-full lg:w-[25%] flex flex-col flex-shrink-0 gap-4"
        >
          <div
            class="flex gap-2 flex-wrap"
          >
            <div
              aria-label="electric"
              class="flex w-fit items-center gap-2 text-white px-3 py-2 rounded-full bg-pokemon-electric p-4 rounded-full transition hover:scale-110"
              data-testid="poke-badge"
            >
              <div
                class="w-6 h-6 flex items-center justify-center rounded-full overflow-visible brightness-[10]"
              >
                <img
                  alt="electric"
                  height="13"
                  src="/icons/png/filled/electric.png"
                  width="13"
                />
              </div>
            </div>
          </div>
          <div
            class="flex flex-col gap-4"
          >
            <h2
              class="text-3xl font-bold drop-shadow-2xl"
            >
              Base Stats:
            </h2>
            <div
              class="flex flex-wrap gap-2 border-l-4 border-white/40 px-4"
            >
              <p
                class="text-xs w-fit shadow-lg font-semibold uppercase text-center text-gray-700 bg-white rounded-xl py-2 px-5 hover:bg-gray-100"
              >
                HP: 35
              </p>
              <p
                class="text-xs w-fit shadow-lg font-semibold uppercase text-center text-gray-700 bg-white rounded-xl py-2 px-5 hover:bg-gray-100"
              >
                Attack: 55
              </p>
              <p
                class="text-xs w-fit shadow-lg font-semibold uppercase text-center text-gray-700 bg-white rounded-xl py-2 px-5 hover:bg-gray-100"
              >
                Defense: 40
              </p>
              <p
                class="text-xs w-fit shadow-lg font-semibold uppercase text-center text-gray-700 bg-white rounded-xl py-2 px-5 hover:bg-gray-100"
              >
                SP. Attack: 50
              </p>
              <p
                class="text-xs w-fit shadow-lg font-semibold uppercase text-center text-gray-700 bg-white rounded-xl py-2 px-5 hover:bg-gray-100"
              >
                SP. Defense: 50
              </p>
              <p
                class="text-xs w-fit shadow-lg font-semibold uppercase text-center text-gray-700 bg-white rounded-xl py-2 px-5 hover:bg-gray-100"
              >
                Speed: 90
              </p>
            </div>
          </div>
        </div>
      </section>
      <div
        class="w-fit flex justify-center self-center md:flex-row lg:flex-col lg:absolute lg:top-0 lg:right-0"
      >
        <div
          class="relative flex items-center py-4 w-full max-w-full overflow-hidden flex-row justify-center lg:flex-col lg:h-screen md:h-auto"
        >
          <button
            class="flex items-center justify-center gap-2 px-4 py-2 rounded-lg bg-blue-600 text-white transition disabled:opacity-50 disabled:cursor-not-allowed px-2 py-2 bg-transparent text-white rounded-md hover:scale-110 disabled:opacity-50 hover:text-red-500 transition"
            data-testid="previous-button"
            disabled=""
          >
            <span>
              ChevronLeft Icon
              ChevronLeft Icon
            </span>
          </button>
          <div
            class="flex space-x-1 md:flex-row lg:flex-col md:space-x-2 lg:space-x-0 md:space-y-0 lg:space-y-2 overflow-hidden"
          >
            <button
              class="flex items-center justify-center gap-2 px-4 py-2 rounded-lg bg-blue-600 text-white transition disabled:opacity-50 disabled:cursor-not-allowed px-4 py-2 rounded-md transition bg-transparent !scale-125 font-bold"
              data-testid="pagination-button-1"
            >
              <span>
                1
              </span>
            </button>
            <button
              class="flex items-center justify-center gap-2 px-4 py-2 rounded-lg bg-blue-600 text-white transition disabled:opacity-50 disabled:cursor-not-allowed px-4 py-2 rounded-md transition bg-transparent text-white hover:scale-125 disabled:opacity-50"
              data-testid="pagination-button-2"
            >
              <span>
                2
              </span>
            </button>
            <button
              class="flex items-center justify-center gap-2 px-4 py-2 rounded-lg bg-blue-600 text-white transition disabled:opacity-50 disabled:cursor-not-allowed px-4 py-2 rounded-md transition bg-transparent text-white hover:scale-125 disabled:opacity-50"
              data-testid="pagination-button-3"
            >
              <span>
                3
              </span>
            </button>
            <button
              class="flex items-center justify-center gap-2 px-4 py-2 rounded-lg bg-blue-600 text-white transition disabled:opacity-50 disabled:cursor-not-allowed px-4 py-2 rounded-md transition bg-transparent text-white hover:scale-125 disabled:opacity-50"
              data-testid="pagination-button-4"
            >
              <span>
                4
              </span>
            </button>
            <button
              class="flex items-center justify-center gap-2 px-4 py-2 rounded-lg bg-blue-600 text-white transition disabled:opacity-50 disabled:cursor-not-allowed px-4 py-2 rounded-md transition bg-transparent text-white hover:scale-125 disabled:opacity-50"
              data-testid="pagination-button-5"
            >
              <span>
                5
              </span>
            </button>
            <button
              class="flex items-center justify-center gap-2 px-4 py-2 rounded-lg bg-blue-600 text-white transition disabled:opacity-50 disabled:cursor-not-allowed px-4 py-2 rounded-md transition bg-transparent text-white hover:scale-125 disabled:opacity-50"
              data-testid="pagination-button-6"
            >
              <span>
                6
              </span>
            </button>
            <button
              class="flex items-center justify-center gap-2 px-4 py-2 rounded-lg bg-blue-600 text-white transition disabled:opacity-50 disabled:cursor-not-allowed px-4 py-2 rounded-md transition bg-transparent text-white hover:scale-125 disabled:opacity-50"
              data-testid="pagination-button-7"
            >
              <span>
                7
              </span>
            </button>
            <button
              class="flex items-center justify-center gap-2 px-4 py-2 rounded-lg bg-blue-600 text-white transition disabled:opacity-50 disabled:cursor-not-allowed px-4 py-2 rounded-md transition bg-transparent text-white hover:scale-125 disabled:opacity-50"
              data-testid="pagination-button-8"
            >
              <span>
                8
              </span>
            </button>
            <button
              class="flex items-center justify-center gap-2 px-4 py-2 rounded-lg bg-blue-600 text-white transition disabled:opacity-50 disabled:cursor-not-allowed px-4 py-2 rounded-md transition bg-transparent text-white hover:scale-125 disabled:opacity-50"
              data-testid="pagination-button-9"
            >
              <span>
                9
              </span>
            </button>
            <button
              class="flex items-center justify-center gap-2 px-4 py-2 rounded-lg bg-blue-600 text-white transition disabled:opacity-50 disabled:cursor-not-allowed px-4 py-2 rounded-md transition bg-transparent text-white hover:scale-125 disabled:opacity-50"
              data-testid="pagination-button-10"
            >
              <span>
                10
              </span>
            </button>
            <button
              class="flex items-center justify-center gap-2 px-4 py-2 rounded-lg bg-blue-600 text-white transition disabled:opacity-50 disabled:cursor-not-allowed px-4 py-2 rounded-md transition bg-transparent text-white hover:scale-125 disabled:opacity-50"
              data-testid="pagination-button-11"
            >
              <span>
                11
              </span>
            </button>
            <button
              class="flex items-center justify-center gap-2 px-4 py-2 rounded-lg bg-blue-600 text-white transition disabled:opacity-50 disabled:cursor-not-allowed px-4 py-2 rounded-md transition bg-transparent text-white hover:scale-125 disabled:opacity-50"
              data-testid="pagination-button-12"
            >
              <span>
                12
              </span>
            </button>
            <button
              class="flex items-center justify-center gap-2 px-4 py-2 rounded-lg bg-blue-600 text-white transition disabled:opacity-50 disabled:cursor-not-allowed px-4 py-2 rounded-md transition bg-transparent text-white hover:scale-125 disabled:opacity-50"
              data-testid="pagination-button-13"
            >
              <span>
                13
              </span>
            </button>
            <button
              class="flex items-center justify-center gap-2 px-4 py-2 rounded-lg bg-blue-600 text-white transition disabled:opacity-50 disabled:cursor-not-allowed px-4 py-2 rounded-md transition bg-transparent text-white hover:scale-125 disabled:opacity-50"
              data-testid="pagination-button-14"
            >
              <span>
                14
              </span>
            </button>
            <button
              class="flex items-center justify-center gap-2 px-4 py-2 rounded-lg bg-blue-600 text-white transition disabled:opacity-50 disabled:cursor-not-allowed px-4 py-2 rounded-md transition bg-transparent text-white hover:scale-125 disabled:opacity-50"
              data-testid="pagination-button-15"
            >
              <span>
                15
              </span>
            </button>
          </div>
          <button
            class="flex items-center justify-center gap-2 px-4 py-2 rounded-lg bg-blue-600 text-white transition disabled:opacity-50 disabled:cursor-not-allowed px-2 py-2 bg-transparent text-white rounded-md disabled:opacity-50 hover:text-red-500 transition"
            data-testid="next-button"
          >
            <span>
              ChevronDown Icon
              ChevronLeft Icon
            </span>
          </button>
        </div>
      </div>
    </div>
  </section>
</div>
`;
